packages/relay-compiler/codegen/RelayCodeGenerator.js

```diff
@@ -53,6 +53,16 @@ function generate(node: Batch | Fragment): RequestNode | ConcreteFragment {
 }
 
 const RelayCodeGenVisitor = {
+  enter(node) {
+    if (node.kind === 'Fragment' || node.kind === 'LinkedField') {
+      const selections = flattenArray(node.selections);
+      const idFieldSelection = selections.find(
+        selection => selection.metadata && selection.metadata.isDataID,
+      );
+      node.idField = idFieldSelection && idFieldSelection.name;
+    }
+    return node;
+  },
```

Comment:
@josephsavona I think in your retracted initial review you said that this should become its own transformer, right? I think that makes sense, it’s kinda awkward here.

Reply:
Yeah this should be its own transform rather than happening at codegen time

Reply:
Will do 👍